scripts/ensure-venv.sh

```diff
@@ -2,40 +2,51 @@
 
 # optionally opt out of virtualenv creation
 # WARNING: this will be removed (most likely renamed) soon!
-if [ "$SENTRY_NO_VIRTUALENV_CREATION" == "1" ]; then
+if [[ "$SENTRY_NO_VIRTUALENV_CREATION" == "1" ]]; then
     exit 0
 fi
 
+yellow="$(tput setaf 3)"
+bold="$(tput bold)"
+reset="$(tput sgr0)"
+
 venv_name=".venv"
-python_bin="python2.7"
+python_version="3.6"
 
-if [ "$SENTRY_PYTHON3" = "1" ]; then
-    venv_name=".venv3"
-    python_bin="python3.6"
+if [[ "$SENTRY_PYTHON2" = "1" ]]; then
+    venv_name=".venv2"
+    python_version="2.7"
 fi
 
-die() {
-    cat >&2 "${@}"
+die () {
+    cat <<EOF
+$@
+EOF
     exit 1
 }
 
-if [ -n "$VIRTUAL_ENV" ]; then
-    # we're enforcing that virtualenv be in .venv, since future tooling e.g. venv-update will rely on this.
-    if [ "$VIRTUAL_ENV" != "${PWD}/${venv_name}" ]; then
-        die "You're in a virtualenv, but it's not in the expected location (${PWD}/${venv_name})"
-    fi
-
-    # TODO: Update this to strictly check .python-version
-    if [ "$SENTRY_PYTHON3" = "1" ]; then
-        python -c "import sys; sys.exit(sys.version_info[:2] != (3, 6))" ||
-            die "For some reason, the virtualenv isn't Python 3.6."
+if [[ -n "$VIRTUAL_ENV" ]]; then
+    major=`python -c "import sys; print(sys.version_info[0])"`
+    minor=`python -c "import sys; print(sys.version_info[1])"`
+    # If .venv contains Python2 or SENTRY_PYTHON2 is set, then fail with instructions
+    if [[ "$major" -eq 2 ]] || [[ ! -z "$SENTRY_PYTHON2" ]]; then
+        cat << EOF
+${yellow}${bold}
+WARNING! You are running a Python 2 virtualenv, which is DEPRECATED.
+Support will be dropped beginning in 2021.
+${reset}
+EOF
     else
-        python -c "import sys; sys.exit(sys.version_info[:2] != (2, 7))" ||
-            die "For some reason, the virtualenv isn't Python 2.7."
+        # If .venv is less than Python 3.6 fail
+        [[ "$minor" -lt 6 ]] &&
```

Comment:
You don't have to fix this since this is all just temporary transitional code, but I want to point out here that this is the Python 4 problem again.